test/ipns.js

```diff
@@ -0,0 +1,265 @@
+          (cb) => goDaemon1.stop(cb),
+          (cb) => goDaemon2.stop(cb)
+        ], done)
+      })
+    })
+
+    // Error: failed to find any peer in table
+    /*
+    series([
+      (cb) => goDf.spawn({
+        repoPath: dir,
+        initOptions: { bits: 1024 },
+      }, (err, node) => {
+        expect(err).to.not.exist()
+        goDaemon1 = node
+        cb()
+      }),
+      (cb) => goDaemon1.start(cb),
+      (cb) => goDaemon1.api.id((err, res) => {
+        expect(err).to.not.exist()
+        goId1 = res
+        cb()
+      }),
+      (cb) => goDaemon1.api.name.publish(ipfsRef, { resolve: false }, cb),
+      (cb) => {
+        goDaemon1.api.name.resolve(jsId, (err, res) => {
+          expect(err).to.not.exist()
+          expect(res).to.equal(ipfsRef)
+          cb()
+        })
+      },
+      (cb) => goDaemon1.stop(cb),
+    ], done) */
+  })
+
+  it('should publish an ipns record to a js daemon and resolve it using a go daemon through the reuse of the same repo', function (done) {
+    this.timeout(100 * 1000)
+
+    const dir = path.join(os.tmpdir(), hat())
+    const ipfsRef = '/ipfs/QmPFVLPmp9zv5Z5KUqLhe2EivAGccQW2r7M7jhVJGLZoZU'
+
+    let goDaemon1
+    let goDaemon2
+    let jsDaemon
+    let jsId
+
+    series([
+      (cb) => jsDf.spawn({
+        repoPath: dir,
+        disposable: false,
+        initOptions: { bits: 512 }
+      }, (err, node) => {
+        expect(err).to.not.exist()
+        jsDaemon = node
+        jsDaemon.init(cb)
+      }),
+      (cb) => jsDaemon.start(cb),
+      (cb) => jsDaemon.api.id((err, res) => {
+        expect(err).to.not.exist()
+        jsId = res.id
+        cb()
+      }),
+      (cb) => jsDaemon.api.name.publish(ipfsRef, { resolve: false }, cb),
+      (cb) => jsDaemon.stop(cb),
+      (cb) => setTimeout(cb, 2000),
+      (cb) => goDf.spawn({
+        repoPath: dir,
+        // disposable: false, // SHOULD BE NON DISPOSABLE BUT THE START FAILS (CHECK repo.js)
+        initOptions: { bits: 1024 }
+      }, (err, node) => {
+        expect(err).to.not.exist()
+        goDaemon1 = node
+        cb()
+      }),
+      (cb) => goDf.spawn({ initOptions: { bits: 1024 } }, (err, node) => {
+        expect(err).to.not.exist()
+        goDaemon2 = node
+        cb()
+      }),
+      (cb) => goDaemon1.start(cb),
+      (cb) => goDaemon2.start(cb),
+      (cb) => goDaemon1.api.id((err, res) => {
+        expect(err).to.not.exist()
+        console.log('diff id', res.id, res.id === jsId) // ID different because of non disposable
+        cb()
+      }),
+      (cb) => {
+        goDaemon1.api.name.resolve(jsId, (err, res) => {
+          console.log('err', err) // Different repo
+          console.log('res', res)
+          cb()
+        })
+      },
+      (cb) => goDaemon1.stop(cb),
+      (cb) => goDaemon2.stop(cb)
+    ], done)
+  })
+
+  it('should publish an ipns record to a go daemon and resolve it using a js daemon through the reuse of the same repo', function (done) {
```

Comment:
In this test, I can use the same repo (id is equal for both), but cannot resolve the record. I am not sure if it is a problem of the JS implementation, or the datastore. I would like to debug the GO part, but is is running in an EXEC file. Any suggestion for debugging in the go side?

I will analyze this deeper, in order to understand if this a problem with the IPNS implementation, or with the repo.